### DISCUSSION TITLE
Regarding the UI zooming update

### DISCUSSION THREAD

**Discussion by SnailPerson** [bitbof/klecks#108] [Ideas]
First off, im very glad it finally is possible to zoom into stuff, that's a huge boost to QOL.
However the dialogue windows are still quite small, so even if you CAN zoom in, its difficult to make out what you are doing in a larger scale due to the small window.
In comparison to the much bigger text dialogue window, stuff like adjust saturation or editing are tiny!
If they too get the size of the current text tool window, that would be perfect!

**Comment by bitbof** [comment]
That will definitely happen!

**Comment by bitbof** [comment]
Increased the size of the updated ones. Long term it will be more responsive and probably larger.

**Reply by SnailPerson** [reply]
will "transform" get zoomer update too? 🥺

**Reply by bitbof** [reply]
Yes. Very soon.

**Reply by SnailPerson** [reply]
:D

**Reply by bitbof** [reply]
released! (0.6.6)

**Reply by SnailPerson** [reply]
awesome! Actually a random thought that popped into my head when i saw the new hand and pencil icon in the transform window, wouldn't it be theoretically possible to merge the perspective function and the transform function to one?
Like say you have an additional button, which also is a dropdown menu for both "before" and "after" functions:
![2023_12_18_10l_Kleki](https://github.com/bitbof/klecks/assets/97041048/eeceeaf4-de2e-4d84-a256-222008a60fa0)
![2023_12_18_10s_Kleki](https://github.com/bitbof/klecks/assets/97041048/072aedfe-79cc-432a-8709-8a40b2b4f8c4)

idk this is just a random idea

**Reply by bitbof** [reply]
Interesting! Would you like something like this because perspective is lacking features? (e.g. in transform it recognizes the bounds, and you can drag the entire thing)

In the long run I want to get rid of these dialogs, and have it all happen in the select/transform tool on the main canvas. Things would then live in the same spot. And ideally with an option to warp.

**Reply by SnailPerson** [reply]
Ooh warping would be very useful for photoshop stuffs!
I suggested this because if its easier to do more stuff during an ongoing change / perhaps its required to resize and adjust perspective for e.g. a background. you'd have to switch between two dialogue windows constantly, so this would be essentially QoL.

Speaking of QoL, the new zooming feature resets every time you leave the dialogue so for stuff like color changes you'd have to zoom in again and again. Im having this issue right now...
could it perhaps be possible to have the zoom level saved for the dialogue windows?
Doesnt have to be per dialogue, just a general "you left with this zoom, welcome back with the same zoom".

Also thanks for responding so quickly and so openly taking your communities suggestions into consideration ^-^

**Reply by bitbof** [reply]
I see!

I'm a bit concerned that remembering zoom can get confusing. I do see the annoyance though and will keep it in mind.

**Reply by SnailPerson** [reply]
perhaps a setting to turn it on for those who need it? The default would stay Alzheimers dialogue.
